src/main/java/io/nats/client/JetStream.java

```diff
@@ -13,15 +13,91 @@
 
 package io.nats.client;
 
+import io.nats.client.api.ConsumerConfiguration;
 import io.nats.client.api.PublishAck;
+import io.nats.client.api.StorageType;
+import io.nats.client.api.StreamConfiguration;
 import io.nats.client.impl.Headers;
 
 import java.io.IOException;
 import java.time.Duration;
 import java.util.concurrent.CompletableFuture;
 
 /**
- * JetStream context for creation and access to streams and consumers in NATS.
+ * JetStream context for access to streams and consumers in NATS.
+ * 
+ * <h3>Basic usage</h3>
+ * 
+ * <p>{@link #publish(String, byte[]) JetStream.Publish} will send a message on the specified subject as a Req/Rep waiting for acknowledgement. 
+ * An error will be received if no stream is litetning on said subject.
+ * 
+ * <p>{@link Connection#publish(String, byte[]) Connection.Publish} can be used as well to send to a stream on a subject, but no confirmation or error will come back. 
+ * Use non acknowledged publish with caution as it is easy to overwhelm Jetstream this way.
+ * 
+ * <p>{@link #publishAsync(String, byte[]) PublishAsync} will not wait for acknowledgement but return a {@link CompletableFuture CompletableFuture}, 
+ * which can be checked for acknowledgement at a later point.
+ * 
+ * <p> Use {@link #getStreamContext(String ) getStreamContext(String)} for <b>consuming/subscribing</b> messages from Jetstream. 
+ * It is <b>recommened</b> to manage consumers explicitely through {@link StreamContext StreamContext} or {@link JetStreamManagement JetStreamManagement}
+ * 
+ * <p>{@link #subscribe(String)} is a convenience method for implicitely creating a consumer on a stream and receiving messages. 
+ * This is simpler but eventually more prone to configuration errors than the {@link ConsumerContext ConsumerContext} based subscription. We recommend to manage consumers explcitely.
+ * 
+ * <h3>Creating Streams, consumer and publishing</h3>
+ * <pre>
+ *  io.nats.client.Connection nc = Nats.connect();
+ *		
+ *  JetStreamManagement jsm = nc.jetStreamManagement();
+ *  StreamConfiguration sc = StreamConfiguration.builder()
+ *    .name("my-stream")
+ *    .storageType(StorageType.File)
+ *    .subjects("foo.*", "bar.*")
+ *    .build();
+ *      
+ *  jsm.addStream(sc);
+ *       
+ *  ConsumerConfiguration consumerConfig = ConsumerConfiguration.builder()
+ *    .durable("my-consumer")
+ *    .build();
+ *		
+ *  jsm.createConsumer("my-stream", consumerConfig);
+ *    	
+ *  io.nats.client.JetStream js = nc.jetStream();
+ *  ConsumerContext consumerContext = js.getConsumerContext("my-stream", "my-consumer"); 
+ *	consumerContext.consume(
```

Comment:
I think any example of consume needs to show
```
MessageConsumer mc = consumerContext.consume
```

I'm not entirely sure if se should add an example here, maybe can we reference an example or something instead.

Reply:
Usually I want ready to run examples outside of the API docs. We have an issue with Jetstream though as the recommended usage pattern is NOT obvious. 
I think this will the biggest example we will have. In all other place it should be back to 3-5 line code fragments

Reply:
If we move the examples, I want to keep a big big link back to the Github location. And at least a few lines of pseudo code to hint at the usage pattern and then link saying: "See full code eaxmples here"

Reply:
A short pseudo code version could be:

//For full example see: <Github> link

Connection nc = Nats.connect();
//Setting up a stream and a consumer	
JetStreamManagement jsm = nc.jetStreamManagement();

StreamConfiguration sc = StreamConfiguration.builder() ......
jsm.addStream(sc);
       
ConsumerConfiguration consumerConfig = ConsumerConfiguration.builder()  ....
 jsm.createConsumer("my-stream", consumerConfig);
     	
//Listening and publishing  
JetStream js = nc.jetStream();

ConsumerContext consumerContext = js.getConsumerContext("my-stream", "my-consumer"); 
MessageConsumer mc = consumerContext.consume(  new Handler( .... ) );
  	
 js.publish("foo.joe", "Hello World".getBytes());